docs/distribution/README.md

```diff
@@ -0,0 +1,566 @@
+
+### Signature Persistence - Signing Multi-arch Manifests
+
+Taking the above scenario further, a signature can be associated with an individual manifest, or a signature can be applied to an index. The index could be a multi-arch index (windows & linux), or the index might represent a [CNAB][cnab].
+
+In the below case, the `net-monitor` software is available as windows (`net-monitor:v1-win`) and linux (`net-monitor:v1-lin`) images, as well as a multi-arch index (`net-monitor:v1`)
+The platform specific images, and the multi-arch index are all signed by **wabbit-networks** and **acme-rockets**.
+
+<img src="../../media/signature-as-index-signing-multi-arch-index.png" width=1100>
+
+- Objects (1-3) are equivalent to above references, with the exception that (1) is changed from a platform specific manifest to a multi-arch index
+- Objects (4-5) represent architecture specific manifests for the multi-arch manifest (1)
+- Objects (6-9) are Notary v2 signatures by the originating author (**wabbit-networks**) and the consumer (**acme-rockets**)
+
+## Signature Discovery
+
+The [OCI distribution-spec][oci-distribution] describes the action of [pushing content][oci-distribution-push] and [pulling of content][oci-distribution-pull].  Pulling a manifest and the associated layers implies a registry must store some linkage between the manifest and its references to layers and config. There are implied additional references between an [OCI Index][oci-index] and its referenced manifest.
+
+To support the [Notary v2 workflow][notaryv2-workflow], where the system knows of the artifact being referenced (1), but doesn't know what signatures might exist on that artifact (2-3, 6-9), a discovery API is required to return the objects that refer to the target artifact (1).
+
+Similar to pulling of an artifact, the referrer API implies a reverse lookup is possible. Based on a given artifact digest, what other objects are referencing that object.
+
+To generalize discovery, a `referrer-metadata` API is proposed to enable discovery of an referenced objects. To support this reverse lookup prototype, additions are proposed to the Notary v2 fork of the reference implementation [docker/distribution][notaryv2-distribution] through [notaryv2-referrer-api].
+
+A referrer is any registry artifact that has an immutable reference to a manifest. An OCI index is a referrer to each manifest it references. The [OCI image spec][oci-image] does not include a config property for an OCI index and there is no reverse lookup of referrers in docker distribution.
+
+A modified OCI index with an `index.config` property that references a collection of manifests allows us to associate a "type" to the referrer-referenced relationship, where the `index.config.mediaType` = `application/vnd.cncf.notary.config.v2+jwt`.
+
+### referrer-metadata API
+
+```HTTP
+`GET http://localhost:5000/v2/net-monitor/manifests/11wma2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49a11wm/referrer-metadata`

```

Comment:
From API's perspective, IMHO, it would be a better design if we first introduce the `referrer` API, which returns all referrer, including the optionally selected metadata, such as the digest of config blob.
The caller may filter the result by adding `config-mediaType` to the query string.